add test examples for thunks and components

tmkelly28 opened the pull request.

collin commented:
I actually think the substance of this PR is :+1: and wouldn't block it for my comments re: formatting/linting.

tmkelly28 commented:
@collin point taken, though (and no time like the present for changes) - I've added instructions to the README for installing eslint, and given a recommendation for how to use it (as well as pointing out that the existing code is written with Standard)

And I removed the errant semicolon...
